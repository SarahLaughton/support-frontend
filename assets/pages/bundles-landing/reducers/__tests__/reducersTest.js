// @flow

// ----- Imports ----- //

import type { Contrib, Amount } from 'helpers/contributions';

import reducer from '../reducers';


// ----- Tests ----- //

jest.mock('ophan', () => {});

describe('reducer tests', () => {

  it('should return the initial state', () => {

    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_TYPE to ONE_OFF', () => {

    const contribType: Contrib = 'ONE_OFF';
    const action = {
      type: 'CHANGE_CONTRIB_TYPE',
      contribType,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toEqual(contribType);
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_TYPE to MONTHLY', () => {

    const contribType: Contrib = 'MONTHLY';
    const action = {
      type: 'CHANGE_CONTRIB_TYPE',
      contribType,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toEqual(contribType);
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_TYPE to ANNUAL', () => {

    const contribType: Contrib = 'ANNUAL';
    const action = {
      type: 'CHANGE_CONTRIB_TYPE',
      contribType,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toEqual(contribType);
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_AMOUNT', () => {

    const amount: Amount = {
      value: '50',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT',
      amount,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toMatchSnapshot();
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount.monthly).toEqual(amount);
    expect(newState.amount.oneOff).toEqual(amount);
  });

  it('should handle CHANGE_CONTRIB_AMOUNT_MONTHLY', () => {

    const amount: Amount = {
      value: '45',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT_MONTHLY',
      amount,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toMatchSnapshot();
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount.monthly).toEqual(amount);
    expect(newState.amount.oneOff).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_AMOUNT_ANNUAL', () => {

    const amount: Amount = {
      value: '50',
      userDefined: false,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT_ANNUAL',
      amount,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toMatchSnapshot();
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount.annual).toEqual(amount);
    expect(newState.amount.oneOff).toMatchSnapshot();
  });

  it('should handle CHANGE_CONTRIB_AMOUNT_ONEOFF', () => {

    const amount: Amount = {
      value: '45',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT_ONEOFF',
      amount,
    };

    const newState = reducer(undefined, action);

    expect(newState.type).toMatchSnapshot();
    expect(newState.error).toMatchSnapshot();
    expect(newState.amount.oneOff).toEqual(amount);
    expect(newState.amount.monthly).toMatchSnapshot();
  });
});
