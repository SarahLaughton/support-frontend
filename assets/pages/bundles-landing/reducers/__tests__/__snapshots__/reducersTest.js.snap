// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT 1`] = `"MONTHLY"`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT 2`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ANNUAL 1`] = `"MONTHLY"`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ANNUAL 2`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ANNUAL 3`] = `
Object {
  "userDefined": false,
  "value": "50",
}
`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_MONTHLY 1`] = `"MONTHLY"`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_MONTHLY 2`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_MONTHLY 3`] = `
Object {
  "userDefined": false,
  "value": "50",
}
`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ONEOFF 1`] = `"MONTHLY"`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ONEOFF 2`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_AMOUNT_ONEOFF 3`] = `
Object {
  "userDefined": false,
  "value": "10",
}
`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to ANNUAL 1`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to ANNUAL 2`] = `
Object {
  "annual": Object {
    "userDefined": false,
    "value": "75",
  },
  "monthly": Object {
    "userDefined": false,
    "value": "10",
  },
  "oneOff": Object {
    "userDefined": false,
    "value": "50",
  },
}
`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to MONTHLY 1`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to MONTHLY 2`] = `
Object {
  "annual": Object {
    "userDefined": false,
    "value": "75",
  },
  "monthly": Object {
    "userDefined": false,
    "value": "10",
  },
  "oneOff": Object {
    "userDefined": false,
    "value": "50",
  },
}
`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to ONE_OFF 1`] = `null`;

exports[`reducer tests should handle CHANGE_CONTRIB_TYPE to ONE_OFF 2`] = `
Object {
  "annual": Object {
    "userDefined": false,
    "value": "75",
  },
  "monthly": Object {
    "userDefined": false,
    "value": "10",
  },
  "oneOff": Object {
    "userDefined": false,
    "value": "50",
  },
}
`;

exports[`reducer tests should return the initial state 1`] = `
Object {
  "amount": Object {
    "annual": Object {
      "userDefined": false,
      "value": "75",
    },
    "monthly": Object {
      "userDefined": false,
      "value": "10",
    },
    "oneOff": Object {
      "userDefined": false,
      "value": "50",
    },
  },
  "error": null,
  "type": "MONTHLY",
}
`;
